app/helpers/petition_helper.rb

```diff
@@ -32,6 +32,18 @@ def waiting_for_debate_date_in_words(petition)
     end
   end
 
+  def public_petition_facets_with_counts(petition_search)
+    Hash[
+      public_petition_facets.map do |key|
+        if petition_search.facets.key? key.to_sym
+          [key, petition_search.facets[key.to_sym]]
+        else
+          nil
+        end
+      end.compact
+    ]
+  end
```

Comment:
If you implement `slice` on `Browseable::Facets` you can simplify this down, e.g:

``` ruby
module Browseable
  class Facets
    def slice(*keys)
      keys = keys.map(&:to_sym)

      keys.each_with_object({}) do |key, hash|
        hash[keys] = self[key] if has_key?(key)
      end
    end
  end
end

@petitions.facets.slice(public_facets)
```